addons/account/models/product.py

```diff
@@ -59,16 +67,17 @@ class ProductTemplate(models.Model):
 
     def _get_product_accounts(self):
         return {
-            'income': self.property_account_income_id or self.categ_id.property_account_income_categ_id,
-            'expense': self.property_account_expense_id or self.categ_id.property_account_expense_categ_id
+            'income': (
+                self.property_account_income_id
+                or self.categ_id.property_account_income_categ_id
+                or self.env.company.income_account_id
```

Comment:
```suggestion
                or (self.company_id or self.env.company).income_account_id
```
if the product belongs to a specific company, we should use that specific company and not the environment one.